### PR TITLE
samples: new sample bigquery_auth_drive_scope

### PR DESCRIPTION
**Pull request by stephaniewang526**
Fixes b/146925135

### CODE REVIEW

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+                ImmutableSet.of(
+                    "https://www.googleapis.com/auth/bigquery",
+                    "https://www.googleapis.com/auth/drive"));
+
+    // Instantiate a client.
+    BigQuery bigquery =
+        BigQueryOptions.newBuilder().setCredentials(credentials).build().getService();
+
+    // Use the client.
+    System.out.println("Datasets:");
```
Comment (kurtisvg) [inline samples/src/main/java/com/example/bigquery/AuthDriveScope.java]: Doesn't really make sense to print this multiple times. 
```suggestion
    System.out.println("Auth succeeded with multiple scopes. Datasets:");
```

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    for (Dataset dataset : bigquery.listDatasets().iterateAll()) {
+      System.out.printf(
+          "Auth succeeded with multiple scopes. Dataset %s%n", dataset.getDatasetId().getDataset());
```
Comment (kurtisvg) [inline samples/src/main/java/com/example/bigquery/AuthDriveScope.java]:
```suggestion
          "Dataset: %s%n", dataset.getDatasetId().getDataset());
```

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+    // Create credentials with Drive & BigQuery API scopes.
+    // Both APIs must be enabled for your project before running this code.
+    GoogleCredentials credentials =
+        ServiceAccountCredentials.getApplicationDefault()
+            .createScoped(
+                ImmutableSet.of(
+                    "https://www.googleapis.com/auth/bigquery",
+                    "https://www.googleapis.com/auth/drive"));
+
+    // Instantiate a client.
```
Comment (kurtisvg) [inline samples/src/main/java/com/example/bigquery/AuthDriveScope.java]: Can we use the wording on the other samples? About reusing the client for multiple requests?

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+    BigQuery bigquery =
+        BigQueryOptions.newBuilder().setCredentials(credentials).build().getService();
+
+    // Use the client.
+    System.out.println("Datasets:");
+    for (Dataset dataset : bigquery.listDatasets().iterateAll()) {
+      System.out.printf(
+          "Auth succeeded with multiple scopes. Dataset %s%n", dataset.getDatasetId().getDataset());
+    }
+  }
```
Comment (kurtisvg) [inline samples/src/main/java/com/example/bigquery/AuthDriveScope.java]: Shouldn't this catch the `BigQueryException` error?

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java (hunk 3 above)
Comment (stephaniewang526) [reply samples/src/main/java/com/example/bigquery/AuthDriveScope.java]: yes - will do

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java (hunk 4 above)
Comment (stephaniewang526) [reply samples/src/main/java/com/example/bigquery/AuthDriveScope.java]: yes - agree

File: samples/src/main/java/com/example/bigquery/AuthDriveScope.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+          BigQueryOptions.newBuilder().setCredentials(credentials).build().getService();
+
+      // Use the client.
+      System.out.println("Auth succeeded with multiple scopes. Datasets:");
+      for (Dataset dataset : bigquery.listDatasets().iterateAll()) {
+        System.out.printf("Dataset: %s%n", dataset.getDatasetId().getDataset());
+      }
+    } catch (BigQueryException e) {
+      System.out.println("Auth failed due to error: \n" + e.toString());
+    }
```
Comment (lesv) [inline samples/src/main/java/com/example/bigquery/AuthDriveScope.java]: Ok, but if you aren't handling it, and not providing any useful advice, you probably should let the higher level handle.

**Pull request merged**